packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -371,27 +371,32 @@ class FlutterDriverExtension {
     return null;
   }
 
-  /// Returns a future that waits until no pending frame is scheduled (frame is synced).
+  /// Returns a future that waits until the app is idle.
   ///
   /// Specifically, it checks:
-  /// * Whether the count of transient callbacks is zero.
-  /// * Whether there's no pending request for scheduling a new frame.
+  /// * Whether there's no pending [MethodChannel.invokeMethod] calls.
+  /// * Whether the count of transient callbacks is zero, if "frame sync" is enabled.
+  /// * Whether there's no pending request for scheduling a new frame, if "frame sync" is enabled.
   ///
-  /// We consider the frame is synced when both conditions are met.
+  /// We consider the app idle when the conditions listed above are met.
   ///
   /// This method relies on a Flutter Driver mechanism called "frame sync",
   /// which waits for transient animations to finish. Persistent animations will
   /// cause this to wait forever.
   ///
   /// If a test needs to interact with the app while animations are running, it
-  /// should avoid this method and instead disable the frame sync using
-  /// `set_frame_sync` method. See [FlutterDriver.runUnsynchronized] for more
-  /// details on how to do this. Note, disabling frame sync will require the
-  /// test author to use some other method to avoid flakiness.
-  Future<Result> _waitUntilNoPendingFrame(Command command) async {
-    await _waitUntilFrame(() {
-      return SchedulerBinding.instance.transientCallbackCount == 0
-          && !SchedulerBinding.instance.hasScheduledFrame;
+  /// should instead disable the frame sync using the `set_frame_sync` method.
+  /// See [FlutterDriver.runUnsynchronized] for more details on how to do this.
```

Comment:
Should we document on runUnsynchronized that it changes the behavior of waitUntilIdle?

Reply:
On second thought, isn't this interaction kinda wired? Why does runUnsynchronized change the definition of what "idle" means? With this implementation, waiting for idle while running unsynchronized will in fact not wait for the app to be idle...

Reply:
Not sure if I understand your question here... I think runUnsyncrhonized will disable "frame sync"?

Reply:
With the current implementation, if you execute a waitUntilIdle command within runUnsynchronized you are not waiting until the app is idle. That seems odd to me given the method name. So, I was wondering why you decided to make waitUntilIdle depend on set_frame_sync.

Reply:
OK. Got your question.

IMHO, I think it should be fine to call this waitUntilIdle within runsynchronized and it waits until resources are idle except the frame sync. And if an app has animations running all the time, but developers want to test other parts of the app, they have the option to disable frame sync but call this waitUntilIdle to ensure everything else is idle.

On the other hand, since Driver/extension API has the notion of "frame sync" mechanism, I think it makes sense to respect that in a waitUntilIdle API. Otherwise, these APIs act inconsistently: _waitForElement API respects frame sync, but this one doesn't.